lib/manageiq/api/common/headers.rb

```diff
@@ -0,0 +1,44 @@
+module HeaderScope
+  thread_mattr_accessor :current
+end
+
+module ManageIQ
+  module API
+    module Common
+      module Headers
+
+        def self.current=(val)
+          validate_headers(val)
+          HeaderScope.current = val
+        end
+
+        def self.current
+          HeaderScope.current
+        end
+
+        def self.decode(headers, key)
+          validate_headers(headers)
+          val = headers.instance_variable_get(:@req).stringify_keys
+          JSON.parse(Base64.decode64(val[key]))
+        end
+
+        def self.encode(val, key)
+          if val.is_a?(Hash)
+            hashed = val.stringify_keys
+            Base64.strict_encode64(hashed[key].to_json)
+          else
+            raise StandardError, "Must be a Hash"
+          end
+        end
+
+        def self.validate_headers(val)
+          if val.is_a?(ActionDispatch::Http::Headers)
```

Comment:
@syncrou If this is for x-rh-auth-identity should we check if he decoded value conforms to the IPP 

```
{
  "identity": {
    "account_number": "0369233",
    "type": "User",
    "user" : {
      "username": "jdoe",
      "email": "jdoe@acme.com",
      "first_name": "John",
      "last_name": "Doe",
      "is_active": true,
      "is_org_admin": false,
      "is_internal": false,
      "locale": "en_US"
    },
    "internal" : {
      "org_id": "3340851",
      "auth_type": "basic-auth",
      "auth_time": 6300
     }
  }
}

```

Reply:
I'd like to discuss this more.  I originally was thinking ( and I think @Fryguy was too? ) we would have 2 separate modules.  One strictly for raw headers, and a `User` variant that validates the `x-rh-identity` values like what you're suggesting (among the other `.` notation attributes it would offer as well.) I do agree we need a mechanism to sanity check if the `x-rh-identity` key is valid.

Reply:
Yes a separate module should have the header processing.  Similar to madhu comment above, I don't think any Base64 decoding belongs here